A couple tweaks to codejail config

timmc-edx opened the pull request:
Two commits, see descriptions for details:

- Rename apparmor profile
- Allow some file writes

----

I've completed each of the following or determined they are not applicable:

- [x] Made a plan to communicate any major developer interface changes (or N/A)

nedbat commented:
@clintonb @jbarciauskas to match the changes in #113.

clintonb commented on README.rst:
```diff
@@ -38,7 +38,8 @@ Getting Started
 All of the services can be run by following the steps below. Note that since we
 are running many containers, you should configure Docker with a sufficient
 amount of resources. Our testing found that `configuring Docker for Mac`_ with
-2 CPUs and 4GB of memory works well.
```
Per the discussion on #113:
...with a *minimum* of 2 CPUs and 4GB works well.

I'm not sure how to properly express using half of the system as a recommendation.

clintonb replied:
P.S. We can leave out the half system bits for now. I think it's most important that troubleshooting details get merged sooner rather than waiting to rephrase system specs.

nedbat commented:
OK, I think I updated it to the text you two agreed on.

timmc-edx committed:
feat!: Rename codejail apparmor profile to add `openedx` prefix

AppArmor profiles are global within the OS, so we should use a name that is
better guaranteed to not conflict. Choosing an openedx prefix also gives
operators a hint when they're trying to manage installed profiles in the OS
and establishes precedent for future profiles.

timmc-edx committed:
fix: Supply remaining process limit for codejail

- Specify `FSIZE` so that codejail executions can write a small amount
  (default is no writing to disk at all)
- Move `REALTIME` up to be next to `CPU`

robrap commented on codejail.profile:
```diff
@@ -60,7 +60,7 @@ profile codejail_service flags=(mediate_deleted) {
     # Allow sending a kill signal to the codejail_sandbox subprofile when the execution
     # runs beyond time limits.
-    signal (send) set=(kill) peer=codejail_service//codejail_sandbox,
+    signal (send) set=(kill) peer=openedx_codejail_service//codejail_sandbox,
```
Are there bugs that we missed where `codejail_sandbox` references need to be updated?

timmc-edx replied:
Not following -- I don't think there are any functional changes in the renaming part of this diff.

robrap replied:
Oops. I got confused. That said, can you better explain why you wanted the `openedx_` prefix here, and why you don't want it for `openedx_codejail_sandbox`?

timmc-edx replied:
Oh, the idea there is just that the namespacing is only important on the outermost profile. The inner one only shows up in the context of the outer one; it doesn't really have an independent existence. To illustrate, here's what I see when I ask AppArmor about its current profiles (and processes) and filter it to just codejail-related things:

```
$ sudo aa-status | grep codejail
   openedx_codejail_service
   openedx_codejail_service//codejail_sandbox
```
